### PR TITLE
add nix flake

### PR DESCRIPTION
**Pull request by unrooted**
as title says, this PR adds nix flake to this project 

tested - it builds on my machine 
image

### CODE REVIEW

**Comment by arthomas** [comment]
@unrooted Thank you for this PR! 
Upon testing with `nix build` and running the resulting binary (`result/bin/deject`) I received the following error:
```
result/bin/deject                                                                                                                                                                                                                                                                                                                    ✔  29s  
Traceback (most recent call last):
  File "/nix/store/881f1b5d84cjwxmn5h6a28gr3xl89s3p-python3.11-deject-0.4.0/bin/.deject-wrapped", line 6, in <module>
    from deject.main import app
  File "/nix/store/881f1b5d84cjwxmn5h6a28gr3xl89s3p-python3.11-deject-0.4.0/lib/python3.11/site-packages/deject/main.py", line 7, in <module>
    import rzpipe
  File "/nix/store/f96a38ha79djrn5g2cnvggrbnpmvh5f7-python3.11-rzpipe-0.6.0/lib/python3.11/site-packages/rzpipe/__init__.py", line 37, in <module>
    from .open_sync import open
  File "/nix/store/f96a38ha79djrn5g2cnvggrbnpmvh5f7-python3.11-rzpipe-0.6.0/lib/python3.11/site-packages/rzpipe/open_sync.py", line 16, in <module>
    from .open_base import OpenBase
  File "/nix/store/f96a38ha79djrn5g2cnvggrbnpmvh5f7-python3.11-rzpipe-0.6.0/lib/python3.11/site-packages/rzpipe/open_base.py", line 25, in <module>
    from .native import RzCore
  File "/nix/store/f96a38ha79djrn5g2cnvggrbnpmvh5f7-python3.11-rzpipe-0.6.0/lib/python3.11/site-packages/rzpipe/native.py", line 17, in <module>
    lib = CDLL(lib_name)
          ^^^^^^^^^^^^^^
  File "/nix/store/w4fvvhkzb0ssv0fw5j34pw09f0qw84w8-python3-3.11.7/lib/python3.11/ctypes/__init__.py", line 376, in __init__
    self._handle = _dlopen(self._name, mode)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
OSError: librz_core.so.0.6: cannot open shared object file: No such file or directory
```
This was fixed by placing the import in `main.py` for rzpipe in a try/except block:
```
try:
    import rzpipe
except OSError:
    pass
``` 
However, this was not needed when just running with Poetry or Python previously.
I don't mind if this has to be added (Rizin support is still being worked on and would be best to not make it a requirement.)

Thanks!

**Comment by unrooted** [comment]
for now I added it in the `try/except` block 
it builds, so I guess it's good for now, but please check if anything else is needed or if anything else fails on your end

image

**Comment by arthomas** [comment]
Hi @unrooted,

Thanks for the updated PR! Unfortunately, when running `deject run -t pe <file>` I am getting the following errors for Magic and Chepy: 
```
ModuleNotFoundError: No module named 'chepy'
AttributeError: module 'magic' has no attribute 'from_buffer'
```
Unsure why Chepy is not found in this instance. Any ideas?
For Magic, it might be because it's libmagic vs https://pypi.org/project/python-magic/?

Thanks!

**Comment by unrooted** [comment]
hi @arthomas 
updated flake.nix so it should build with no issues as of right now (had to bump m2crypto and typer versions
however, the issue with `magic` still persists - I will take a look at possible solution soon

File: nix/default.nix [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+      radare2
+      ssdeep
+      util-linux
+      file
+
+      m2crypto
+      exceptiongroup
+      hexdump
+      kaitaistruct
+      magic
```
Comment (arthomas) [inline nix/default.nix]:
```suggestion
```

File: nix/default.nix [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  ssdeep,
+  openssl,
+  util-linux,
+  file,
+  # python
+  m2crypto,
+  exceptiongroup,
+  hexdump,
+  kaitaistruct,
+  magic,
```
Comment (arthomas) [inline nix/default.nix]:
```suggestion
```

**Review by arthomas — APPROVED** [review]

**Comment by arthomas** [comment]
Looks good to me.

**Pull request merged**